src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -231,19 +236,13 @@ private void setupFieldPanels() {
         addRequiredTab(type);
 
         // optional fields
-        List<String> displayedOptionalFields = new ArrayList<>();
         Set<String> deprecatedFields = new HashSet<>(EntryConverter.FIELD_ALIASES_TEX_TO_LTX.keySet());
         Set<String> usedOptionalFieldsDeprecated = new HashSet<>(deprecatedFields);
 
         if ((type.getOptionalFields() != null) && !type.getOptionalFields().isEmpty()) {
             if (!frame.getCurrentBasePanel().getBibDatabaseContext().isBiblatexMode()) {
-                displayedOptionalFields.addAll(type.getOptionalFields());
-
                 addOptionalTab(type);
             } else {
-                displayedOptionalFields.addAll(type.getPrimaryOptionalFields());
```

Comment:
Items of the list are never used, only added to it (`displayedOptionalFields` was only used in line 234, 240, 244, and 245; or am I missing something?)